### PR TITLE
fix/73-Toasts

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
Add AllowMultiLineTruncation() in UseTrimbleModus() to set ellips basded on truncation Modify AppBuilderExtensions moved outside of Popups #73

### CODE REVIEW

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs [hunk 1]
```diff
@@ -73,5 +74,32 @@ public static MauiAppBuilder ConfigurePopups(this MauiAppBuilder builder, Action
                 });
             return builder;
         }
+        public static void AllowMultiLineTruncation()
+        {
+            static void UpdateMaxLines(Microsoft.Maui.Handlers.LabelHandler handler, ILabel label)
+            {
+#if ANDROID
```
Comment (rthanga1) [inline Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:81]: Can you also check in Windows? We might (or maynot) probably need something like this for the windows too.

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿using Microsoft.Maui.LifecycleEvents;
 using Trimble.Modus.Components.Popup.Pages;
 
-namespace Trimble.Modus.Components.Popup.Hosting
```
Comment (rthanga1) [inline Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:4]: Build fails because of this

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:81]: Yes will check

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:4]: Replaced with appropriate namespace

**Review by rthanga1 — APPROVED** [review]

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs (hunk 1 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:81]: Is this also done?

Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:81]: I tried for windows but it doesn't work as it does for iOS and android ,but tail truncate works hence it adds ellipsis to the end of the first line , if the message we pass is entered with 3 lines (eg. line1/n line2/n line 3...) it adds ellipsis
To the third line

Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:81]: The Tablet's Toast view is not replicated in the desktop version as there is no specifics given for it, the ellipsis in windows is not set as expected after three lines its added to end of each line .
Screenshot_1685950806
desktop_screen
desktopToast

**Review by rthanga1 — DISMISSED** [review]

File: Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs [hunk 3]
```diff
@@ -146,7 +146,7 @@ private void PopupData(string message, string actionButtonText, Action action)
 
         var idiom = Device.Idiom;
         setWidth(idiom);
-        Message = GetWrappedLabelText(message, idiom);
+        Message = message;
     }
 
     private void setWidth(TargetIdiom idiom)
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs:152]: image
Github suggest to use DeviceIdiom can you please try that

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs [hunk 4]
```diff
@@ -73,5 +74,32 @@ public static MauiAppBuilder ConfigurePopups(this MauiAppBuilder builder, Action
                 });
             return builder;
         }
+        public static void AllowMultiLineTruncation()
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs:77]: Can you please move this method as label handler

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Hosting/AppBuilderExtensions.cs]: Done

File: Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs:152]: Is Resloved in another PR #101

**Review by Sofiya-kumar — APPROVED** [review]

**Pull request merged**